schema-nft-marketplace.graphql

```diff
@@ -26,236 +26,239 @@ enum Network {
+  id: ID!
 
-    " Name of the NFT marketplace, for example LooksRare "
-    name: String!
+  " Name of the NFT marketplace, for example LooksRare "
+  name: String!
 
-    " The blockchain network this subgraph is indexing on. "
-    network: Network!
+  " The blockchain network this subgraph is indexing on. "
+  network: Network!
 
-    " Version of the subgraph schema, in SemVer format (e.g. 1.0.0) "
-    schemaVersion: String!
+  " Version of the subgraph schema, in SemVer format (e.g. 1.0.0) "
+  schemaVersion: String!
 
-    " Version of the subgraph implementation, in SemVer format (e.g. 1.0.0) "
-    subgraphVersion: String!
+  " Version of the subgraph implementation, in SemVer format (e.g. 1.0.0) "
+  subgraphVersion: String!
 
-    " Version of the methodology used to compute metrics, loosely based on SemVer format (e.g. 1.0.0) "
-    methodologyVersion: String!
+  " Version of the methodology used to compute metrics, loosely based on SemVer format (e.g. 1.0.0) "
+  methodologyVersion: String!
 
-    " Number of collections listed on the marketplace. "
-    collectionCount: Int!
+  " Number of collections listed on the marketplace. "
+  collectionCount: Int!
 
-    " Trade count of the all collections on the marketplace. "
-    tradeCount: Int!
+  " Trade count of the all collections on the marketplace. "
+  tradeCount: Int!
 
-    " Cumulative trade volume (in ETH) "
-    cumulativeTradeVolumeETH: BigDecimal!
+  " Cumulative trade volume (in ETH) "
+  cumulativeTradeVolumeETH: BigDecimal!
 
-    " Revenue that goes to the marketplace protocol, aka protocol fee. "
-    marketplaceRevenueETH: BigDecimal!
+  " Revenue that goes to the marketplace protocol, aka protocol fee. "
+  marketplaceRevenueETH: BigDecimal!
 
-    " Revenue that goes to creator, aka royalty fee. "
-    creatorRevenueETH: BigDecimal!
+  " Revenue that goes to creator, aka royalty fee. "
+  creatorRevenueETH: BigDecimal!
 
-    " Sum of marketplaceRevenueETH and creatorRevenueETH. "
-    totalRevenueETH: BigDecimal!
+  " Sum of marketplaceRevenueETH and creatorRevenueETH. "
+  totalRevenueETH: BigDecimal!
 
-    " Cumulative unique traders "
-    cumulativeUniqueTraders: Int!
+  " Cumulative unique traders "
+  cumulativeUniqueTraders: Int!
 }
 
 type Collection @entity {
-    " Contract address. "
-    id: ID!
+  " Contract address. "
+  id: ID!
 
-    " Collectio name, mirrored from the smart contract. Leave null if not available. "
-    name: String
+  " Collectio name, mirrored from the smart contract. Leave null if not available. "
+  name: String
 
-    " Collection symbol, mirrored from the smart contract. Leave null if not available. "
-    symbol: String
+  " Collection symbol, mirrored from the smart contract. Leave null if not available. "
+  symbol: String
 
-    " Total supply of the collection, mirrored from the smart contract. "
-    totalSupply: BigInt!
+  " Total supply of the collection, mirrored from the smart contract. "
+  totalSupply: BigInt
```

Comment:
set optional since `totalSupply` is not part of the ERC721/ERC1155 standard - it only exists in extension interfaces